lib/streamlit/runtime/caching/hashing.py

```diff
@@ -402,15 +396,38 @@ def _to_bytes(self, obj: Any) -> bytes:
         elif isinstance(obj, Enum):
             return str(obj).encode()
 
-        elif type_util.is_type(obj, "pandas.core.frame.DataFrame") or type_util.is_type(
-            obj, "pandas.core.series.Series"
-        ):
+        elif type_util.is_type(obj, "pandas.core.series.Series"):
             import pandas as pd
 
+            h = hashlib.new("md5")
+            self.update(h, obj.size)
```

Comment:
nit: why do we need this for series and not for dataframe?

Reply:
good catch, `df.shape` also added to dataframe hashing